erizo_controller/common/semanticSdp/SDPInfo.js

```diff
@@ -0,0 +1,823 @@
+    }
+    const medias = [];
+    this.medias.forEach((media) => {
+      if (media.getType().toLowerCase() === type.toLowerCase()) {
+        medias.push(media);
+      }
+    });
+    return medias;
+  }
+
+  getMediaById(msid) {
+    let result;
+    this.medias.forEach((media) => {
+      if (media.getId().toLowerCase() === msid.toLowerCase()) {
+        result = media;
+      }
+    });
+    return result;
+  }
+
+  getVersion() {
+    return this.version;
+  }
+
+  getDTLS() {
+    return this.dtls;
+  }
+
+  setDTLS(dtlsInfo) {
+    this.dtls = dtlsInfo;
+  }
+
+  getICE() {
+    return this.ice;
+  }
+
+  setICE(iceInfo) {
+    this.ice = iceInfo;
+  }
+
+  addCandidate(candidate) {
+    this.candidates.push(candidate);
+  }
+
+  addCandidates(candidates) {
+    candidates.forEach((candidate) => {
+      this.addCandidate(candidate);
+    });
+  }
+
+  getCandidates() {
+    return this.candidates;
+  }
+
+  getStream(id) {
+    return this.streams.get(id);
+  }
+
+  getStreams() {
+    return this.streams;
+  }
+
+  getFirstStream() {
+    if (this.streams.values().length > 0) {
+      return this.streams.values()[0];
+    }
+    return null;
+  }
+
+  addStream(stream) {
+    this.streams.set(stream.getId(), stream);
+  }
+
+  removeStream(stream) {
+    this.streams.delete(stream.getId());
+  }
+
+  toJSON() {
+    const sdp = {
+      version: 0,
+      media: [],
+    };
+
+    sdp.version = this.version || 0;
+    sdp.origin = this.origin || {
+      username: '-',
+      sessionId: (new Date()).getTime(),
+      sessionVersion: this.version,
+      netType: 'IN',
+      ipVer: 4,
+      address: '127.0.0.1',
+    };
+
+    sdp.name = this.name || 'semantic-sdp';
+
+    sdp.connection = this.getConnection();
+    sdp.timing = this.timing || { start: 0, stop: 0 };
+
+    let ice = this.getICE();
+    if (ice) {
+      if (ice.isLite()) {
+        sdp.icelite = 'ice-lite';
+      }
+      sdp.iceOptions = ice.getOpts();
+      sdp.iceUfrag = ice.getUfrag();
+      sdp.icePwd = ice.getPwd();
+    }
+
+    sdp.msidSemantic = this.msidSemantic || { semantic: 'WMS', token: '*' };
+    sdp.groups = [];
+
+    const bundle = { type: 'BUNDLE', mids: [] };
+    let dtls = this.getDTLS();
+    if (dtls) {
+      sdp.fingerprint = {
+        type: dtls.getHash(),
+        hash: dtls.getFingerprint(),
+      };
+
+      sdp.setup = Setup.toString(dtls.getSetup());
+    }
+
+    this.medias.forEach((media) => {
+      const md = {
+        type: media.getType(),
+        port: 9,
+        protocol: 'UDP/TLS/RTP/SAVPF',
+        fmtp: [],
+        rtp: [],
+        rtcpFb: [],
+        ext: [],
+        bandwidth: [],
+        candidates: [],
+        ssrcGroups: [],
+        ssrcs: [],
+        rids: [],
+      };
+
+      md.direction = Direction.toString(media.getDirection());
+
+      md.rtcpMux = 'rtcp-mux';
+
+      md.connection = media.getConnection();
+
+      md.xGoogleFlag = media.getXGoogleFlag('conference');
```

Comment:
`getXGoogleFlag`  in MediaInfo does not take a parameter. If we need to check for the `conference` flag in particular, I'd implement that in this PR